src/compas/geometry/curves/nurbs.py

```diff
@@ -0,0 +1,462 @@
+    def dimension(self):
+        raise NotImplementedError
+
+    @property
+    def domain(self):
+        raise NotImplementedError
+
+    @property
+    def order(self):
+        return self.degree + 1
+
+    @property
+    def start(self):
+        raise NotImplementedError
+
+    @property
+    def end(self):
+        raise NotImplementedError
+
+    @property
+    def is_closed(self):
+        raise NotImplementedError
+
+    @property
+    def is_periodic(self):
+        raise NotImplementedError
+
+    @property
+    def is_rational(self):
+        raise NotImplementedError
+
+    @property
+    def bounding_box(self):
+        raise NotImplementedError
+
+    @property
+    def length(self):
+        raise NotImplementedError
+
+    # ==============================================================================
+    # Methods
+    # ==============================================================================
+
+    def copy(self):
+        """Make an independent copy of the current curve."""
+        return NurbsCurve.from_parameters(
+            self.points,
+            self.weights,
+            self.knots,
+            self.multiplicities,
+            self.degree,
+            self.is_periodic
+        )
+
+    def transform(self, T):
+        """Transform this curve."""
+        raise NotImplementedError
+
+    def transformed(self, T):
+        """Transform a copy of the curve."""
+        copy = self.copy()
+        copy.transform(T)
+        return copy
+
+    def space(self, n=10):
+        """Compute evenly spaced parameters over the curve domain."""
+        u, v = self.domain
+        return linspace(u, v, n)
+
+    def xyz(self, n=10):
+        """Compute point locations corresponding to evenly spaced parameters over the curve domain."""
+        return [self.point_at(param) for param in self.space(n)]
+
+    def locus(self, resolution=100):
+        """Compute the locus of all points on the curve.
+
+        Parameters
+        ----------
+        resolution : int
+            The number of intervals at which a point on the
+            curve should be computed. Defaults to 100.
+
+        Returns
+        -------
+        list
+            Points along the curve.
+        """
+        return self.xyz(resolution)
+
+    def point_at(self, u):
+        """Compute a point on the curve.
+
+        Parameters
+        ----------
+        u : float
+            The value of the curve parameter. Must be between 0 and 1.
+
+        Returns
+        -------
+        Point
+            the corresponding point on the curve.
+        """
+        raise NotImplementedError
+
+    def tangent_at(self, u):
+        """Compute the tangent vector at a point on the curve.
+
+        Parameters
+        ----------
+        u : float
+            The value of the curve parameter. Must be between 0 and 1.
+
+        Returns
+        -------
+        Vector
+            The corresponding tangent vector.
+
+        """
+        pass
```

Comment:
why `pass` and not `NotImplementedError`?

Reply:
should be the latter indeed